### PR TITLE
Clarify how Rails associations should be ordered

### PR DESCRIPTION
**Pull request by gylaz**
There are two options how the current wording can be interpreted. This changes
the wording to favor the B option.

```rb
# A
has_many :apples
belongs_to :tree
has_many :watermelons

# B
belongs_to :tree
has_many :apples
has_many :watermelons
```

### CODE REVIEW

**Comment by gabebw** [comment]
I agree with this.

**Comment by jferris** [comment]
Do we want an example to go along with this?

**Comment by gylaz** [comment]
Example added.

File: style/rails/sample.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class SomeClass
+  belongs_to :tree
+  has_many :apples
+  has_many :watermelons
+end
+
```
Comment (gabebw) [inline style/rails/sample.rb]: extra line

File: style/rails/README.md [hunk 2]
```diff
@@ -8,7 +8,8 @@ Rails
 * Name time columns (referring to a time of day with no date) with `_time`
   suffixes.
 * Name initializers for their gem name.
-* Order ActiveRecord associations alphabetically by attribute name.
+* Order ActiveRecord associations alphabetically by association type, then
+  attribute name. [Example][order-associations]
```
Comment (gabebw) [inline style/rails/README.md]: What if we put the link on " Order ActiveRecord associations alphabetically" so the link flows as part of the text?

**Comment by dgalarza** [comment]
:+1:

**Comment by teoljungberg** [comment]
:+1:

File: style/rails/README.md (hunk 2 above)
Comment (gylaz) [inline style/rails/README.md]: I'm not opposed to that idea, thought that would be inconsistent with how we've been providing examples so far. I'm okay with it if we decided to do examples in this manner going forth.

Comment (gabebw) [inline style/rails/README.md]: Let's keep it consistent! Therefore, let's put a period after "Example".

**Pull request merged**